src/api.ts

```diff
@@ -108,4 +117,59 @@ export const getOpenCommitLatestVersion = async (): Promise<
   }
 };
 
+function generatePrefix(): string | undefined {
+  const prefix = config?.OCO_PREFIX
+
+  if (prefix === undefined) {
+    return undefined;
+  }
+
+
+  // Prefix is limited to /*/ regex expressions
+  const prefixIsRegexString = prefix.match(/\/.*\/*/)
+
+  if (prefixIsRegexString) {
+    try {
+      return generatePrefixFromRegex(prefix);
+    } catch (error) {
+      console.error(`Failed to generate prefix from regex: ${error}`);
+      return undefined;
+    }
+  }
+
+  return prefix;
+}
+
 export const api = new OpenAi();
+
+
+function generatePrefixFromRegex(regex: string): string | undefined {
+
+  // We currently only support regex input from git branch name
+
+  const branch = getCurrentGitBranch();
+
+  if (branch === undefined) {
+    return undefined;
+  }
+
+  const regexWithoutSlashes = regex.slice(1, -1);
+  const regexObject = new RegExp(regexWithoutSlashes);
+  const match = branch.match(regexObject);
+
+  if (match === null) {
+    return undefined;
+  }
+
+  return match.length > 1 ? match[1] : match[0];
+}
+
+function getCurrentGitBranch(): string | undefined {
+  try {
+    const branchName = execaCommandSync('git symbolic-ref --short HEAD').stdout.trim()
```

Comment:
is this going to add the full branch name in the commit like: `#234-some-branch-name commit message text goes here`?

Reply:
No, it will take a regex out of the name. See my first commit on this PR.
Very cool to hear about the hackathon!